tasks/eva_3417/src/main/groovy/eva3417/RemediateLowerCaseRefAlt.groovy

```diff
@@ -0,0 +1,462 @@
+                .collectMany(ssClass -> dbEnv.mongoTemplate.find(query(where("_id").in(newHashList)), ssClass))
+                .flatten()
+
+        return existingHashSVEList
+    }
+
+    void insertSVEWithNewHash(List<SubmittedVariantEntity> sveList, Class ssClass) {
+        List<SubmittedVariantEntity> sveWithNewHashList = new ArrayList<>()
+        for (SubmittedVariantEntity sve : sveList) {
+            // create a new SVE with updated ref, alt and hash
+            SubmittedVariant sveModel = sve.getModel()
+            sveModel.setReferenceAllele(sve.getReferenceAllele().toUpperCase())
+            sveModel.setAlternateAllele(sve.getAlternateAllele().toUpperCase())
+            SubmittedVariantEntity submittedVariantEntity = new SubmittedVariantEntity(sve.getAccession(),
+                    getSVENewHash(sve), sveModel, sve.getVersion())
+
+            sveWithNewHashList.add(submittedVariantEntity)
+        }
+
+        // insert updated SVEs into db
+        dbEnv.mongoTemplate.insert(sveWithNewHashList, ssClass)
+    }
+
+    void processHashCollision(List<SubmittedVariantEntity> hashCollisionSVEList,
+                              Set<SubmittedVariantEntity> alreadyExistingHashSVEList, Class ssClass){
+        // In most cases, we should have only one SVE that has a collision
+        // but there can be cases where we have 2 or more SVE in the batch that has a collision
+        Map<String, List<SubmittedVariantEntity>> mapOfNewHashAndSVEList = hashCollisionSVEList.stream()
+                .collect(Collectors.groupingBy(sve->getSVENewHash(sve), Collectors.toList()))
+        Map<String, SubmittedVariantEntity> mapOfHashAndExistingSVEInDB = alreadyExistingHashSVEList.stream()
+                .collect(Collectors.toMap(sve->sve.getHashedMessage(), sve->sve))
+
+        List<SubmittedVariantEntity> SVEToDiscardList = new ArrayList<>()
+        Map<Long, List<SubmittedVariantEntity>> SVEToMergeMap =  new HashMap<>()
+        List<SubmittedVariantEntity> SVEToInsertList = new ArrayList<>()
+        List<SubmittedVariantEntity> SVEToRemoveList = new ArrayList<>()
+
+        // go through each hash collision one at a time
+        for(Map.Entry<String, List<SubmittedVariantEntity>> entry: mapOfNewHashAndSVEList.entrySet()){
+            // all SVE with same new hash
+            List<SubmittedVariantEntity> sveList = entry.getValue()
+            // get the SVE in DB
+            SubmittedVariantEntity sveInDB = mapOfHashAndExistingSVEInDB.get(entry.getKey())
+
+            //assert all SVEs involved has same RS
+            Set<Long> rsSet = sveList.stream().map(sve->sve.getClusteredVariantAccession())
+                    .collect(Collectors.toSet())
+            rsSet.add(sveInDB.getClusteredVariantAccession())
+
+            if(rsSet.size() > 1){
+                //all sve involved does not have same RS, document SVEs in file - don't process
+                documentInFile(sveList)
```

Comment:
Can you add a log statement if we have rs mismatch ?

Reply:
added